src/tools/isabelle_executor.py

```diff
@@ -406,13 +423,10 @@ def _run_stmt_on_isabelle_server(self, stmt: str) -> None:
             begin_clause = IsabelleExecutor.begin_theory_match.findall(self.buffer)
         elif not self._proof_running:
             self.buffer += stmt + '\n' # Add current line to buffer
-            try:
-                # If the action succeeds, we have a new lemma
-                description = self.pisa_env.step(start_state, self.buffer, end_state, delete_old_state=False, forceTimeout=self.timeout_in_sec)
-                if not description.startswith('Step error:'):
-                    last_thm_details = IsabelleExecutor.theorem_match.findall(self.buffer)
-            except FunctionTimedOut:
-                raise Exception("Error: the action timed out.")
```

Comment:
I had to remove timeouts from the client side because they don’t work properly. Unfortunately, it’s going to be very difficult to pass in the desired timeout to PISA because it requires some complicated Python ↔ Scala interfacing. For now, it suffices to change the timeout parameter manually on line 587 in PisaOS.scala.